src/storage/btree.c

```diff
@@ -5074,6 +5091,57 @@ btree_search_nonleaf_page (THREAD_ENTRY * thread_p, BTID_INT * btid, PAGE_PTR pa
 	  *slot_id = middle;
 	  *child_vpid = non_leaf_rec.pnt;
 
+	  if (page_bounds != NULL)
+	    {
+	      if (!page_bounds->m_is_inf_right_key)
```

Comment:
can all this be moved to page_bounds member function? the code is repeated for each case (c == 0, < 0 si > 0).

Reply:
moved